File: CLF_CBF_QP_python/ECBF_2Dmove/cal_system.py
import numpy as np


def cal_system(x_state):
    x = x_state[0]
    y = x_state[1]
    v = x_state[2]
    theta = x_state[3]

    f = np.array([v * np.cos(theta), v * np.sin(theta), 0, 0]).T
    g = np.array([[0, 0], [0, 0], [1, 0], [0, 1]])
    return f, g

File: CLF_CBF_QP_matlab/Dynamic_barrier/cal_system.m
function [f,g] = cal_system(x_state)
    %dx = f + g*u
    x=x_state(1);
    y=x_state(2);
    v=x_state(3);
    theta=x_state(4);
    
    f=[v*cos(theta);v*sin(theta);0;0];
    g=[0 0;0 0;1 0;0 1];
   
end